File: Modules/SMES/DashBoards/SIE.Web.MES.DashBoard/Reports/FpySettings/Commands/AddSettingCommand.js
SIE.defineCommand('SIE.Web.MES.DashBoard.Reports.FpySettings.Commands.AddSettingCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit", iconCls: "iconfont icon-AddEntity icon-green" },
    onItemCreated: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: model,
                success: function (res) {
                    var data = res.Result;
                    entity.setUpdateDate(data.UpdateDate);
                    entity.setUpdateBy(data.UpdateBy);
                }
            }, me.view);
        }
    },
});

File: Modules/SMES/DashBoards/SIE.Web.MES.DashBoard/Reports/FpySettings/Commands/AddSettingCommand.cs
using SIE.MES.DashBoard.Reports.FpySettings;
using SIE.Web.Command;
using System;

namespace SIE.Web.MES.DashBoard.Reports.FpySettings.Commands
{
    /// <summary>
    /// 产品直通率设置命令
    /// </summary>
    [JsCommand("SIE.Web.MES.DashBoard.Reports.FpySettings.Commands.AddSettingCommand")]
    public class AddSettingCommand : ViewCommand
    {
        /// <summary>
        /// 命令执行
        /// </summary>
        /// <param name="args">参数</param>
        /// <param name="scope">作用域</param>
        /// <returns>返回实体</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var current = new ProductFpySetting();
            current.UpdateDate = DateTime.Now;
            current.UpdateBy = RT.IdentityId;
            return current;
        }
    }
}
